### DISCUSSION TITLE
If you want to include a new data function in TDC, let us know here ⬇️

### DISCUSSION THREAD

**Discussion by kexinhuang12345** [mims-harvard/TDC#23] [New Functions]
Hi!

Feel free to drop any pointers for a new data function that you find may be useful for TDC users here!

We would love to discuss it with you and also work with you to include it in TDC!

**Comment by abearab** [comment]
Suggesting a new data function: **Knowledge Graph Mastery**

In this PR [mims-harvard/TDC#207](https://github.com/mims-harvard/TDC/pull/207), I started a few features that I believe can help to build, handle, explore, and integrate knowledge graphs based on the final PrimeKG data format (i.e. simply a pandas data frame with same column names). [Here](https://github.com/mims-harvard/PrimeKG/pull/12#issuecomment-1741878955) I mentioned a bit more details as part of a discussion with @payalchandak, @ayushnoori, and @marinkaz.

The proposed Knowledge Graphs data function can be useful for: 
1. Future updates of PrimeKG 
2. Building new graphs from other resources and datasets, e.g. https://github.com/mims-harvard/PrimeKG/pull/12
3. Building context-specific knowledge graphs, see "Future directions" in [Machine learning predicts cellular response to genetic perturbation. _Nat Biotechnol_ (2023).](https://doi.org/10.1038/s41587-023-01907-4)

Also, I have already used this functionality for other use cases and can provide more updates moving forward.

___
Note that the data function title suggested by chatGPT haha

<img width="500" alt="image" src="https://github.com/mims-harvard/TDC/assets/53412130/a59349e8-32f8-4a5f-b6dc-a67ff5de6ce6">
